bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewHubHandler.java

```diff
@@ -215,8 +239,14 @@ private synchronized void poll() {
         try {
             logger.debug("Polling for state");
             pollShades();
-            pollScenes();
-            pollSceneCollections();
+            List<Scene> scenes = pollScenes();
+            updateSceneChannels(scenes);
+            createDeprecatedSceneChannels(scenes);
+            List<SceneCollection> sceneCollections = pollSceneCollections();
```

Comment:
Shouldn't this code be within its own method?

Reply:
Which code exactly, all four lines? Why?

Reply:
No. I meant all the whole block of lines that you had added at that point. Because they comprise a separate block of functionality Which is not actually covered under the title of @poll()').

Reply:
Lines 243 thru 249. Maybe into **two** sub functions: updateSceneChannels() and updateScheduledEvents()..

Reply:
Ah, in that way. Well, actually it's unchanged, I just divided polling and channel updating into separate methods to make them more distinct and smaller. Do you think it would be more appropriate to rename the `poll` method in your opinion? We fetch scenes (A), scene collections (B) and scheduled events (C) and I'd prefer to fetch A, update channels for A, fetch B, update channels for B, fetch C and update channels for C. So if fetching automations, for example, would fail for any reason, we would still have the scenes and scene collections updated. Any suggestions much welcome. :)

Reply:
@andrewfg - I have cleaned up the `poll` method as I agree it had become a bit cluttered. Methods are still divided, but responsibility to call data fetching methods has now been propagated to the channel update methods. Also, `createDeprecatedSceneChannels` is now called from `updateSceneChannels`, so now it won't be called that often anymore.